osrdyne/src/drivers/kubernetes.rs

```diff
@@ -15,21 +15,72 @@ use k8s_openapi::{
     },
     apimachinery::pkg::apis::meta::v1::LabelSelector,
 };
+use keda::MetricType;
 use kube::{api::ObjectMeta, Client};
+use log::debug;
 use serde::{Deserialize, Serialize};
-use std::{collections::BTreeMap, future::Future, pin::Pin};
+use std::{
+    collections::{BTreeMap, HashMap},
+    future::Future,
+    pin::Pin,
+};
 use uuid::Uuid;
 
+mod keda;
+
 #[derive(Serialize, Deserialize, Clone, Debug)]
-pub struct AutoscalingOptions {
+pub struct HPAOptions {
     /// The minimum number of replicas
     pub min_replicas: i32,
+
     /// The maximum number of replicas
     pub max_replicas: i32,
```

Comment:
Small question, currently osrdyne deploy an horizontal pod autoscaller with the capabilities to schedule multiple pods for the same infra. Why isn't this scale method good enough for us?

Reply:
Basically, Keda is able to use information from within the RabbitMQ. On the backend it creates an HPA that uses a metrics that is provided by Keda itself (instead of CPU/MEM that we could theorically uses).

The rationale behind the usage of Keda instead of CPU/RAM is that the load average for Core is not a good indicator of a required scaling, the latency in message dequeuing will be a more accurate indicator of the current load since it will directly be related to the latency experience by the users.